metrics/devtools/disqualified_endpoint_reporter.go

```diff
@@ -0,0 +1,74 @@
+package devtools
+
+import (
+	"net/http"
+
+	"github.com/pokt-network/poktroll/pkg/polylog"
+
+	"github.com/buildwithgrove/path/protocol"
+)
+
+type (
+	// ProtocolDataReporter is an interface that provides data about sanctioned endpoints at the protocol level.
+	ProtocolDataReporter interface {
+		// GetTotalServiceEndpointsCount returns the total number of service endpoints for a given service ID.
+		GetTotalServiceEndpointsCount(protocol.ServiceID, *http.Request) (int, error)
+		// HydrateDisqualifiedEndpointsResponse hydrates the disqualified endpoint response with the protocol-specific data.
+		HydrateDisqualifiedEndpointsResponse(protocol.ServiceID, *DisqualifiedEndpointResponse)
+	}
+
+	// QoSDataReporter is an interface that provides data about disqualified endpoints at the QoS level.
+	QoSDataReporter interface {
+		// HydrateDisqualifiedEndpointsResponse hydrates the disqualified endpoint response with the QoS-specific data.
+		HydrateDisqualifiedEndpointsResponse(protocol.ServiceID, *DisqualifiedEndpointResponse)
+	}
+)
+
+// DisqualifiedEndpointReporter is a reporter that collects data about disqualified
+// endpoints from both the protocol and QoS levels.
+//
+// It is used by the `/disqualified_endpoints` URL path in the router to provide
+// useful information about currently disqualified endpoints for development and debugging.
+type DisqualifiedEndpointReporter struct {
```

Comment:
1. Lowercase
2. Enforce struct = interface for compilation type interface support